core/src/main/java/com/valtech/aem/saas/core/http/response/PojoDataExtractionStrategy.java

```diff
@@ -0,0 +1,32 @@
+package com.valtech.aem.saas.core.http.response;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.valtech.aem.saas.core.http.response.dto.FacetCountsDTO;
+import lombok.RequiredArgsConstructor;
+
+import java.util.Optional;
+
+/**
+ * A strategy for extracting facet counts data.
+ */
+@RequiredArgsConstructor
+public final class PojoDataExtractionStrategy<T> implements SearchResponseDataExtractionStrategy<T> {
```

Comment:
Why is this called PojoData? Isn't the idea to always have concrete Classes such as BestBets or Highlight Objects?

Reply:
Yup, I agree with you. My idea is to extract this logic, so we don't repeat it for each and every POJO

Reply:
aha, are you planning to do this in another PR?

Reply:
i'd say yes :) 
Don't want to add more stuff to this one. For now i'll modify the current impl to suit SearchResultItemTrackingExtractionStrategy

Reply:
ok that sounds like a good approach. feel free to create a ticket ;)